lib/drivers/bayerContourNext.js

```diff
@@ -0,0 +1,510 @@
+              annotate.annotateEvent(smbg, ann);
+            });
+          }
+        dataToPost.push(smbg);
+      }
+    }else{
+      debug('Device has no records to upload');
+      throw(new Error('Device has no records to upload'));
+    }
+
+    return dataToPost;
+  };
+
+  return {
+    detect: function(deviceInfo, cb){
+      debug('no detect function needed', arguments);
+      cb(null, deviceInfo);
+    },
+
+    setup: function (deviceInfo, progress, cb) {
+      console.log('in setup!');
+      progress(100);
+      cb(null, {deviceInfo: deviceInfo});
+    },
+
+    connect: function (progress, data, cb) {
+      debug('in connect!');
+
+      cfg.deviceComms.connect(data.deviceInfo, bcnPacketHandler, probe, function(err) {
+        if (err) {
+          return cb(err);
+        }
+        data.disconnect = false;
+        progress(100);
+        cb(null, data);
+
+      });
+    },
+
+    getConfigInfo: function (progress, data, cb) {
+      debug('in getConfigInfo', data);
+
+      getDeviceInfo({}, function (err, result) {
+          progress(100);
+
+          if(!err){
+              data.connect = true;
+              _.assign(data, result);
+
+              cb(null, data);
+          }else{
+              cb(err,result);
+          }
+      });
+    },
+
+    fetchData: function (progress, data, cb) {
+      debug('in fetchData', data);
+
+      function getOneRecordWithProgress(recnum, cb) {
+        var rec = data.rawrecords.shift();
+        progress(100.0 * recnum / data.nrecs);
+        setTimeout(function() {
+          getOneRecord(rec, cb);
+        }, 20);
+      }
+
+      async.timesSeries(data.nrecs, getOneRecordWithProgress, function(err, result) {
+        if (err) {
+          debug('fetchData failed');
+          debug(err);
+          debug(result);
+        } else {
+          debug('fetchData', result);
+        }
+        data.fetchData = true;
+        data.bgmReadings = result;
+        progress(100);
+        cb(err, data);
+      });
+    },
+
+    processData: function (progress, data, cb) {
+        progress(0);
+        data.bg_data = processReadings(data.bgmReadings);
+        data.post_records = prepBGData(progress, data);
+        var ids = {};
+        for (var i = 0; i < data.post_records.length; ++i) {
+            var id = data.post_records[i].time + '|' + data.post_records[i].deviceId;
+            if (ids[id]) {
+                debug('duplicate! %s @ %d == %d', id, i, ids[id] - 1);
+                debug(data.post_records[ids[id] - 1]);
+                debug(data.post_records[i]);
+            } else {
+                ids[id] = i + 1;
+            }
+        }
+        progress(100);
+        data.processData = true;
+        cb(null, data);
+    },
+
+    uploadData: function (progress, data, cb) {
+      progress(0);
+      var sessionInfo = {
+        deviceTags: ['bgm'],
+        deviceManufacturers: ['Bayer'],
+        deviceModel: 'BayerContour Next',
```

Comment:
Just `Contour Next` for deviceModel please, since Bayer is in the manufacturers.